src/Ast.ml

```diff
@@ -1218,10 +1218,9 @@ end = struct
     prec_ast ast
     >>| fun prec_ast ->
     let cmp = Poly.compare prec_ctx prec_ast in
-    if cmp < 0 then (* ast higher precedence than context: no parens *)
+    if cmp < 0 (* ast higher precedence than context: no parens *) then
       false
-    else if cmp > 0 then (* context higher prec than ast: add parens *)
-      true
+    else if cmp > 0 then (* context higher prec than ast: add parens *) true
```

Comment:
These two changes seem to indicate that something isn't quite right.

Reply:
I think this is a consequence of the issue fixed in #253. I locally rebased this PR on #253 and reverted the change to Ast.ml and then it reformats without change.